tests/config.spec.coffee

```diff
@@ -482,79 +483,70 @@ describe 'Device Config:', ->
-								deltaEndpoint: 'https://delta.resin.io'
-								deviceId: 3
-								uuid: '7cf02a62a3a84440b1bb5579a3d57469148943278630b17e7fc6c4f7b465c9'
-								apiKey: '1234'
-								deviceType: 'raspberry-pi'
-								userId: 13
-								listenPort: 48484
-								pubnubSubscribeKey: '5678'
-								pubnubPublishKey: '1234'
-								mixpanelToken: 'asdf'
-								registered_at: 15000
-								username: 'johndoe'
-								appUpdatePollInterval: 60000
-								connectivity: 'connman'
-								files:
-									'network/settings': '''
-										[global]
-										OfflineMode=false
-										TimeUpdates=manual
-
-										[WiFi]
-										Enable=true
-										Tethering=false
-
-										[Wired]
-										Enable=true
-										Tethering=false
-
-										[Bluetooth]
-										Enable=true
-										Tethering=false
-									'''
-									'network/network.config': '''
-										[service_home_ethernet]
-										Type = ethernet
-										Nameservers = 8.8.8.8,8.8.4.4
-									'''
+						config =
+							applicationId: 999
+							applicationName: 'App1'
+							apiEndpoint: 'https://api.resin.io'
+							vpnPort: 1723
+							vpnEndpoint: 'vpn.resin.io'
+							registryEndpoint: 'registry.resin.io'
+							deltaEndpoint: 'https://delta.resin.io'
+							deviceId: 3
+							uuid: '7cf02a62a3a84440b1bb5579a3d57469148943278630b17e7fc6c4f7b465c9'
+							deviceType: 'raspberry-pi'
+							userId: 13
+							listenPort: 48484
+							pubnubSubscribeKey: '5678'
+							pubnubPublishKey: '1234'
+							mixpanelToken: 'asdf'
+							registered_at: 15000
+							username: 'johndoe'
+							appUpdatePollInterval: 60000
+							connectivity: 'connman'
+							files:
+								'network/settings': '''
+									[global]
+									OfflineMode=false
+									TimeUpdates=manual
+
+									[WiFi]
+									Enable=true
+									Tethering=false
+
+									[Wired]
+									Enable=true
+									Tethering=false
+
+									[Bluetooth]
+									Enable=true
+									Tethering=false
+								'''
+								'network/network.config': '''
+									[service_home_ethernet]
+									Type = ethernet
+									Nameservers = 8.8.8.8,8.8.4.4
+								'''
+
+						describe 'given no device api key', ->
+							it 'should eventually become a valid provisioning key configuration', ->
+								promise = deviceConfig.getByDevice('7cf02a62a3a84440b1bb5579a3d57469148943278630b17e7fc6c4f7b465c9', {})
+								m.chai.expect(promise).to.eventually.become _.assign {}, config,
+									apiKey: '1234'
+
+						describe 'given a device api key', ->
+							it 'should eventually become a valid device key configuration', ->
+								promise = deviceConfig.getByDevice('7cf02a62a3a84440b1bb5579a3d57469148943278630b17e7fc6c4f7b465c9', '4321', {})
+								console.log(require('util').inspect(_.assign({}, config, deviceApiKey: '4321'), depth: null))
+								m.chai.expect(promise).to.eventually.become _.assign {}, config,
+									deviceApiKey: '4321'
```

Comment:
It does, there's no `apiKey` property on the shared `config`, so if there's one in the result then the two object won't match (I actually hit an issue whilst writing the tests because `apiKey: undefined` was being returned and that isn't the same as being unset